ckcp/hack/util/reset_ckcp.sh

```diff
@@ -0,0 +1,288 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SCRIPT_DIR="$(
+  cd "$(dirname "$0")" >/dev/null || exit
+  pwd
+)"
+
+PLNSVC_DIR="$(dirname "$(dirname "$(dirname "$SCRIPT_DIR")")")"
+CKCP_DIR="$PLNSVC_DIR/ckcp"
+GITOPS_DIR="$PLNSVC_DIR/gitops"
+CONFIG="$PLNSVC_DIR/config/config.yaml"
+
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
+RESET_HARD="false"
+
+usage() {
+  printf "
+Usage:
+    %s [options]
+
+Scrap ckcp and free resources deployed by openshift_dev_setup.sh script.
+
+Mandatory arguments:
+    --kcp-kubeconfig KCP_KUBECONFIG
+        kubeconfig to the kcp-workspace.
+
+Optional arguments:
+    --reset-hard RESET_HARD
+        setting this option to 'true'
+        will aggresively remove operators deployed as well.
+        Defalut value: \$RESET_HARD
+    -k, --kubeconfig KUBECONFIG
+        kubeconfig to the compute cluster.
+        The current context will be used.
+        Default value: \$KUBECONFIG
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    %s --kcp-kubeconfig './tmp/tmp.435kjkdsf/credentials/kubeconfig/kcp/ckcp.kubeconfig'
+" "${0##*/}" "${0##*/}" >&2
+}
+
+parse_args() {
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+    -k | --kubeconfig)
+      shift
+      KUBECONFIG="$1"
+      ;;
+    --kcp-kubeconfig)
+      shift
+      KCP_KUBECONFIG="$1"
+      ;;
+    --reset-hard)
+      shift
+      RESET_HARD="$1"
+      ;;
+    -d | --debug)
+      set -x
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    --)
+      # End of arguments
+      break
+      ;;
+    *)
+      exit_error "Unknown argument: $1"
+      ;;
+    esac
+    shift
+  done
+}
+
+prechecks() {
+  KCP_KUBECONFIG=${KCP_KUBECONFIG:-}
+  if [[ -z "${KCP_KUBECONFIG}" ]]; then
+    printf "\n\n[ERROR] Missing parameter --kcp-kubeconfig"
+    exit
+  fi
+}
+
+# Removes different Argo CD applications deployed
+uninstall_pipeline_service(){
+    printf "\n  Uninstalling Pipeline Service:\n"
+    # Remove Pipeline-Service Argo CD application
+    kubectl delete -f "$GITOPS_DIR/argocd/argocd.yaml"
+    # Remove all the child applications deployed by Pipeline-Service
+    kubectl delete -k "$GITOPS_DIR/argocd/argo-apps"
+
+    # Check if the Argo CD applications have been indeed removed
+    # list of all the Argo Apps that a user can deploy using Pipeline Service
+    mapfile -t all_argo_apps < <(kubectl kustomize "$GITOPS_DIR/argocd/argo-apps"  | yq '.metadata.name' | grep -e '[a-z]')
+    all_argo_apps+=("pipeline-service")
+    # list of all the Argo Apps still deployed
+    mapfile -t argo_apps_deployed < <(argocd app list -o yaml | yq '.[].metadata.name')
+    matched_apps=()
+    for app in "${all_argo_apps[@]}"
```

Comment:
Please favor the `for ...; do` syntax (single line statement). Occurs multiple times in the file.